applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_max_stress_response_function.cpp

```diff
@@ -0,0 +1,239 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "adjoint_max_stress_response_function.h"
+#include "utilities/compare_elements_and_conditions_utility.h"
+
+namespace Kratos
+{
+    // --------------------------------------------------------------------------
+    AdjointMaxStressResponseFunction::AdjointMaxStressResponseFunction(ModelPart& rAdjointModelPart, Parameters ResponseSettings)
+    : AdjointStructuralResponseFunction(rAdjointModelPart, ResponseSettings),
+      mrAdjointModelPart(rAdjointModelPart),
+      mCriticalPartName(ResponseSettings["critical_part_name"].GetString())
+    {
+        mTracedStressType = StressResponseDefinitions::ConvertStringToTracedStressType(ResponseSettings["stress_type"].GetString());
+        mStressTreatment = StressResponseDefinitions::ConvertStringToStressTreatment(ResponseSettings["stress_treatment"].GetString());
+
+        if(ResponseSettings.Has("echo_level"))
+            mEchoLevel = ResponseSettings["echo_level"].GetInt();
+
+        KRATOS_ERROR_IF(mStressTreatment != StressTreatment::Mean)
+            << "AdjointMaxStressResponseFunction::AdjointMaxStressResponseFunction: Specified stress treatment not supported: " << ResponseSettings["stress_type"].GetString() << std::endl;
+
+        // Set default traced element in adjoint model part (first in the list of agglomerated elements).
+        ModelPart& adjoint_agglomeration_part = mrAdjointModelPart.GetSubModelPart(mCriticalPartName);
+        auto first_element = *(adjoint_agglomeration_part.ElementsBegin());
+        mpTracedElementInAdjointPart = mrAdjointModelPart.pGetElement(first_element.Id());
```

Comment:
this should be nullptr before it is properly assigned in CalculateValue

Reply:
Done. I also added a corresponding error check.